redundant-bmc/src/manager.cpp

```diff
@@ -94,11 +122,20 @@ Role Manager::determineRole()
 
         role_determination::Input input{
             .bmcPosition = services->getBMCPosition(),
+            .previousRole = previousRole,
             .siblingPosition = siblingPosition,
             .siblingRole = siblingRole,
             .siblingHeartbeat = sibling->hasHeartbeat(),
             .siblingProvisioned = siblingProvisioned};
 
+        // If an error case forced it to passive last time, don't use
+        // the previous role in the determination so that we don't try
+        // to choose the role just because that's what was used last time.
+        if (chosePassiveDueToError)
+        {
+            input.previousRole = Role::Unknown;
```

Comment:
On line 75, I now do updateRole(determineRole())

and then in updateRole() starting on line 218, it will set passiveError to the new value, possibly clearing it, so I think that is probably good enough, unless you think otherwise.